master/internal/db/postgres_test_utils.go

```diff
@@ -63,22 +64,48 @@ type Model struct {
 	WorkspaceID     int           `db:"workspace_id" json:"workspace_id"`
 }
 
+func init() {
+	testOnlyDBLock = func(sql *sqlx.DB) (unlock func()) {
+		tx, err := sql.Beginx()
+		if err != nil {
+			panic(err)
+		}
+
+		const lockID = 0x33ad0708c9bed25b // Chosen arbitrarily.
+		if _, err := tx.Exec("SELECT pg_advisory_xact_lock($1)", lockID); err != nil {
+			_ = tx.Rollback()
+			panic(err)
+		}
+
+		return func() {
+			if err := tx.Commit(); err != nil {
+				_ = tx.Rollback()
+				panic(err)
+			}
+		}
+	}
+}
+
 // ResolveTestPostgres resolves a connection to a postgres database. To debug tests that use this
 // (or otherwise run the tests outside of the Makefile), make sure to set
 // DET_INTEGRATION_POSTGRES_URL.
-func ResolveTestPostgres() (*PgDB, error) {
+func ResolveTestPostgres() (*PgDB, func(), error) {
 	pgDB, err := ConnectPostgres(os.Getenv("DET_INTEGRATION_POSTGRES_URL"))
 	if err != nil {
-		return nil, fmt.Errorf("failed to connect to postgres: %w", err)
+		return nil, nil, fmt.Errorf("failed to connect to postgres: %w", err)
 	}
-	return pgDB, nil
+	return pgDB, func() {
+		if err := pgDB.Close(); err != nil {
+			panic(err)
+		}
+	}, nil
 }
 
 // MustResolveTestPostgres is the same as ResolveTestPostgres but with panics on errors.
-func MustResolveTestPostgres(t *testing.T) *PgDB {
-	pgDB, err := ResolveTestPostgres()
+func MustResolveTestPostgres(t *testing.T) (*PgDB, func()) {
```

Comment:
is this an emerging style in go land to return an object + closing/cancellation function? I thought something like

```
db := MustResolveTestPostgres(...)
defer db.Close()
```

would be more idiomatic

Reply:
Yeah I have seen it in some places. I know Ryan used it in ```func ResolveNewPostgresDatabase() (*PgDB, func(), error) {```. I think I have seen it in other established Go APIs but I'm not recalling where. The one advantage I see is it more intentional that you need to close something. Its hard for someone to know if they need to or don't need to call the ```Close()```. 


Doing that, the linter complains that we aren't checking the error. We could add another helper or something. I wanna do another pass on this code to reduce duplication and split tests into isolated db tests soon (so each package would be running in a seperate database instance).

```
internal/db/postgres_rbac_intg_test.go:354:18: Error return value of `pgDB.Close` is not checked (errcheck)
```